tests/shap/test_log.py

```diff
@@ -0,0 +1,93 @@
+        additional_pip_deps=pip_deps,
+        additional_conda_channels=None,
+    )
+
+def test_sklearn_log_explainer():
+    """
+    Tests mlflow.shap log_explainer with mlflow serialization of the underlying model
+    """
+
+    with mlflow.start_run() as run:
+
+        run_id = run.info.run_id
+
+        X, y = shap.datasets.boston()
+        model = sklearn.ensemble.RandomForestRegressor(n_estimators=100)
+        model.fit(X, y)
+    
+        explainer_original = shap.Explainer(model.predict, X, algorithm='permutation')
+        shap_values_original = explainer_original(X[:5])
+
+        mlflow.shap.log_explainer(explainer_original, "test_explainer")
+
+        explainer_new = mlflow.shap.load_explainer("runs:/" + run_id + "/test_explainer")
+        shap_values_new = explainer_new(X[:5])
+
+        assert np.array_equal(shap_values_original.base_values,shap_values_new.base_values)
+        assert type(explainer_original) == type(explainer_new)
+        assert type(explainer_original.masker) == type(explainer_new.masker)
+
+def test_sklearn_log_explainer_self_serialization():
+    """
+    Tests mlflow.shap log_explainer with SHAP internal serialization of the underlying model
+    """
+
+    with mlflow.start_run() as run:
+
+        run_id = run.info.run_id
+
+        X, y = shap.datasets.boston()
+        model = sklearn.ensemble.RandomForestRegressor(n_estimators=100)
+        model.fit(X, y)
+    
+        explainer_original = shap.Explainer(model.predict, X, algorithm='permutation')
+        shap_values_original = explainer_original(X[:5])
+
+        mlflow.shap.log_explainer(explainer_original, "test_explainer", serialize_model_using_mlflow=False)
+
+        explainer_new = mlflow.shap.load_explainer("runs:/" + run_id + "/test_explainer")
+        shap_values_new = explainer_new(X[:5])
+
+        assert np.array_equal(shap_values_original.base_values,shap_values_new.base_values)
+        assert type(explainer_original) == type(explainer_new)
+        assert type(explainer_original.masker) == type(explainer_new.masker)
+
+def test_sklearn_log_explainer_pyfunc():
+    """
+    Tests mlflow.shap log_explainer with mlflow serialization of the underlying model using pyfunc flavor
+    """
+
+    with mlflow.start_run() as run:
+
+        run_id = run.info.run_id
+
+        X, y = shap.datasets.boston()
+        model = sklearn.ensemble.RandomForestRegressor(n_estimators=100)
+        model.fit(X, y)
+    
+        explainer_original = shap.Explainer(model.predict, X, algorithm='permutation')
+        shap_values_original = explainer_original(X[:2])
+
+        mlflow.shap.log_explainer(explainer_original, "test_explainer")
+
+        explainer_pyfunc = mlflow.pyfunc.load_model("runs:/" + run_id + "/test_explainer")
+        shap_values_new = explainer_pyfunc.predict(X[:2])
+        
+
+        assert shap_values_original.values.shape == shap_values_new.shape
```

Comment:
Can we use `np.testing.assert_frame_equal` here isntead?

Reply:
Using np.testing.assert_allclose